### PR TITLE
drivers: hci: stm32_ipm: add UID BLE address

### PR DESCRIPTION
**Pull request by mtuxpe**
The STM32WB has a 64 bit Unique Device Number UID.
The UID is used by firmware to derive 48-bit
Device Address EUI-48.

Signed-off-by: Marcio Montenegro <mtuxpe@gmail.com>

### CODE REVIEW

File: drivers/bluetooth/hci/ipm_stm32wb.c [hunk 1]
```diff
@@ -54,7 +55,19 @@ struct aci_set_tx_power {
...
+struct aci_set_ble_addr {
+	u8_t config_offset;
+	u8_t length;
+	u8_t value[6];
+} __packed;
+
 #define ACI_WRITE_SET_TX_POWER_LEVEL       BT_OP(BT_OGF_VS, 0xFC0F)
+#define ACI_HAL_WRITE_CONFIG_DATA	   BT_OP(BT_OGF_VS, 0xFC0C)
+
+#define CONFIG_DATA_PUBADDR_OFFSET		0
```
Comment (joerchan) [inline drivers/bluetooth/hci/ipm_stm32wb.c]: This does not come from kconfig? I think the CONFIG_ prefix might be reserved.

File: drivers/bluetooth/hci/ipm_stm32wb.c [hunk 2]
```diff
@@ -9,15 +9,16 @@
...
 #include <bluetooth/hci.h>
 #include <bluetooth/hci_driver.h>
+#include "bluetooth/addr.h"
 
 #include "app_conf.h"
 #include "stm32_wpan_common.h"
 #include "shci.h"
 #include "shci_tl.h"
 
+
```
Comment (joerchan) [inline drivers/bluetooth/hci/ipm_stm32wb.c]: Usually we don't have 2 empty lines

File: drivers/bluetooth/hci/ipm_stm32wb.c [hunk 3]
```diff
@@ -357,6 +370,73 @@ static void start_ble_rf(void)
...
+static int bt_ipm_set_addr(void)
+{
+	bt_addr_t *uid_addr;
+	struct aci_set_ble_addr *param;
+	struct net_buf *buf, *rsp;
+	int err;
+
+	uid_addr = bt_get_ble_addr();
+	if (uid_addr == NULL) {
```
Comment (joerchan) [inline drivers/bluetooth/hci/ipm_stm32wb.c]:
```suggestion
	if (!uid_addr) {
```

**Review by joerchan — COMMENTED** [review]
Looks good. Only nitpicking

File: drivers/bluetooth/hci/ipm_stm32wb.c (hunk 1 above)
Comment (mtuxpe) [reply drivers/bluetooth/hci/ipm_stm32wb.c]: No.I think I need to rename this defines

Comment (mtuxpe) [reply drivers/bluetooth/hci/ipm_stm32wb.c]: fixed

File: drivers/bluetooth/hci/ipm_stm32wb.c (hunk 2 above)
Comment (mtuxpe) [reply drivers/bluetooth/hci/ipm_stm32wb.c]: done

File: drivers/bluetooth/hci/ipm_stm32wb.c (hunk 3 above)
Comment (mtuxpe) [reply drivers/bluetooth/hci/ipm_stm32wb.c]: fixed

**Review by joerchan — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by erwango — APPROVED** [review]
Tested ok on nucleo_wb55rg

**Pull request merged**